PWGJE/Tasks/photonChargedTriggerProducer.cxx

```diff
@@ -0,0 +1,347 @@
+    // number global tracks
+    int nGlobalTracks = 0;
+
+    // count global tracks (for independence of multiplicity task (uses only JE derieved data))
+    for (auto const& track : tracks) {
+      // track selection
+      if (!checkGlobalTrackEta(track))
+        continue;
+
+      nGlobalTracks++;
+
+      if (!isSelectedEvent)
+        continue;
+      if (track.pt() < ptTrigMin)
+        continue;
+
+      isTriggerEvent = true;
+
+      // trigger info
+      triggerTable(track.collisionId(), track.globalIndex(), track.pt(), track.phi(), track.eta());
+    }
+
+    // collision info
+    collisionExtraCorrTable(isSelectedEvent, isTriggerEvent, nGlobalTracks);
+  }
+  PROCESS_SWITCH(PhotonChargedTriggerProducer, processRecoCollisionTrigger, "process correlation collision_extra and trigger table (reconstructed)", false);
+
+  void processRecoPipmTPCTOF(aod::JetCollision const& collision,
+                             soa::Join<aod::JetTracks, aod::JTrackPIs> const& tracks, soa::Join<aod::Tracks, aod::TracksExtra, aod::pidTPCPi, aod::pidTOFPi> const&)
+  {
+    // event selection
+    if (!checkEventSelection(collision))
+      return;
+
+    // hadron/pipm
+    for (auto const& track : tracks) {
+      // track selection
+      if (!checkGlobalTrackEta(track))
+        continue;
+
+      // hadron
+      hadronTable(track.collisionId(), track.globalIndex(), track.pt(), track.phi(), track.eta());
+
+      // pipm selection
+      auto const& trackPID = track.track_as<soa::Join<aod::Tracks, aod::TracksExtra, aod::pidTPCPi, aod::pidTOFPi>>();
+      if (!checkPipmTPCTOF(trackPID))
+        continue;
+
+      // pipm
+      pipmTable(track.collisionId(), track.globalIndex(), track.pt(), track.phi(), track.eta());
+    }
+  }
+  PROCESS_SWITCH(PhotonChargedTriggerProducer, processRecoPipmTPCTOF, "process pipm (TPC-TOF) table (reconstructed)", false);
+
+  void processRecoPipmTPC(aod::JetCollision const& collision,
+                          soa::Join<aod::JetTracks, aod::JTrackPIs> const& tracks, soa::Join<aod::Tracks, aod::pidTPCPi> const&)
+  {
+    // event selection
+    if (!checkEventSelection(collision))
+      return;
+
+    // hadron/pipm
+    for (auto const& track : tracks) {
+      // track selection
+      if (!checkGlobalTrackEta(track))
+        continue;
+
+      // hadron
+      hadronTable(track.collisionId(), track.globalIndex(), track.pt(), track.phi(), track.eta());
+
+      // pipm selection
+      auto const& trackPID = track.track_as<soa::Join<aod::Tracks, aod::pidTPCPi>>();
+      if (!checkPipmTPC(trackPID))
+        continue;
+
+      // pipm
+      pipmTable(track.collisionId(), track.globalIndex(), track.pt(), track.phi(), track.eta());
+    }
+  }
+  PROCESS_SWITCH(PhotonChargedTriggerProducer, processRecoPipmTPC, "process pipm (TPC) table (reconstructed)", false);
+
+  void processRecoPhotonPCM(soa::Join<aod::JetCollisions, aod::JCollisionPIs>::iterator const& collision, aod::Collisions const&,
```

Comment:
why do you need aod::Collisions const& here?

Reply:
You still need aod::JCollisionPIs because thats where the Id colum is stored. However the aod::Collisions table is the one you probably dont need to subscribe to